src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1015,21 +1015,27 @@ private Symbol UnwrapAlias(Symbol symbol, out AliasSymbol alias, DiagnosticBag d
             }
             else
             {
+                var boundTypeArguments = BindTypeArguments(typeArguments, diagnostics, basesBeingResolved);
+                if (unconstructedType.IsGenericType && options.IsAttributeTypeLookup() && boundTypeArguments.Any(bta => bta.Type.IsTypeParameter()))
+                {
+                    // error CS0416: 'T': an attribute argument cannot use type parameters
+                    diagnostics.Add(ErrorCode.ERR_AttrArgWithTypeVars, node.Location, boundTypeArguments.First(bta => bta.Type.IsTypeParameter()).ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat));
+                }
                 // It's not an unbound type expression, so we must have type arguments, and we have a 
                 // generic type of the correct arity in hand (possibly an error type). Bind the type 
                 // arguments and construct the final result. 
                 resultType = ConstructNamedType(
                     unconstructedType,
                     node,
                     typeArguments,
-                    BindTypeArguments(typeArguments, diagnostics, basesBeingResolved),
+                    boundTypeArguments,
                     basesBeingResolved,
                     diagnostics);
             }
 
-            if (options.IsAttributeTypeLookup())
+            if (!Compilation.LanguageVersion.AllowGenericAttributes() && options.IsAttributeTypeLookup())
```

Comment:
Similarly, this block should probably be removed. The result of `BindGenericSimpleNamespaceOrTypeOrAliasSymbol` should not depend on LangVer (only diagnostics). We should be able to test that with semantic model (with older and newer LangVer).

Reply:
fixed